projects/igniteui-angular/src/public_api.ts

```diff
@@ -86,6 +87,8 @@ export * from './lib/services/index';
 export * from './lib/core/dates';
 export * from './lib/core/density';
 export { CancelableEventArgs } from './lib/core/utils';
+export { InteractionMode } from './lib/time-picker/time-picker.common';
```

Comment:
Export this in `IgxTimePickerModule` as we will have the same enum for the date picker.